include/tsl/sparse_hash.h

```diff
@@ -2026,6 +2025,11 @@ class sparse_hash: private Allocator, private Hash, private KeyEqual, private Gr
         const slz_size_type nb_deleted_buckets = deserialize_value<slz_size_type>(deserializer);
         const float max_load_factor = deserialize_value<float>(deserializer);
         
+        if(hash_compatible) {
```

Comment:
Instead of adding an extra `if` here, I'd prefer to just move and duplicate the `this->max_load_factor(max_load_factor);` into both branches of the `if(!hash_compatible) { .. } else { ... }` (after `m_bucket_count = ...` in the else).